### DISCUSSION TITLE
Client component renders depending on window.innerWidth

### DISCUSSION THREAD

**Discussion by GuillaumeHalb** [vercel/next.js#67577] [Help]
### Summary

I have the following component that renders either the mobile node or desktop node from props:

```
const Responsive: FC<{
  breakpoint?: number;
  desktop: ReactNode;
  mobile: ReactNode;
}> = ({ breakpoint = 600, desktop, mobile }) => {
  const [isMobile, setIsMobile] = useState<boolean>();

  useEffect(() => {
    const handleResize = () => setIsMobile(window.innerWidth < breakpoint);

    handleResize();

    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
  }, [breakpoint]);

  if (isMobile) {
    return mobile;
  }
  return desktop;
};
```

The first renders occurs **before** the useEffect.
I cannot set the state because window is undefined.

I feel like, even if I'm using a client component here, it first renders server side and then switch to client component.

Is this behaviour expected? I mean, don't we want a client component to be rendered only client side?

Is there a way to set state before the renders occurs?

I don't use css because I use the same component ref in both mobile and desktop.

### Additional information

```js
An easy way to reproduce:

> yarn create next-app


export default function Home() {
  return (
    <main className={styles.main}>
      <Responsive desktop={<div>Desktop</div>} mobile={<div>Mobile</div>} />
    </main>
  );
}
```
```


### Example

_No response_

**Comment by GuillaumeHalb** [comment]
It is very similar to: https://github.com/vercel/next.js/discussions/44886

**Comment by wwwdepot** [comment]
https://nextjs.org/docs/app/api-reference/functions/userAgent
